### PR TITLE
chore(deps): update to sdk v0.50.0-rc.0

### PR DESCRIPTION
**Pull request by mmsqe**
for more [info](https://github.com/cosmos/cosmos-sdk/compare/69db56b49bb7...v0.50.0-rc.0)

### CODE REVIEW

**Comment by itsdevbear** [comment]
lgtm thanks king

**Review by itsdevbear — DISMISSED** [review]

**Comment by itsdevbear** [comment]
@mmsqe can you update the gh actions to use 1.21 as well?

**Comment by itsdevbear** [comment]
@mmsqe and docker SORRY LMAO

**Comment by mmsqe** [comment]
> @mmsqe and docker SORRY LMAO

I'm not quite familiar with the base, review in caution

**Comment by itsdevbear** [comment]
lgtm should be good

**Comment by itsdevbear** [comment]
base is our base container that all other containers use

**Review by itsdevbear — APPROVED** [review]

**Comment by itsdevbear** [comment]
@calbera @ocnc this is probably good now

File: eth/core/state/journal/transient_storage_test.go [hunk 1]
```diff
@@ -46,41 +46,41 @@ var _ = Describe("TransientStorage", func() {
...
 		ts.Finalize()
-		Expect(ts.Size(), 0)
+		Expect(ts.Size()).To(Equal(0))
 		Expect(func() { ts.Finalize() }).ToNot(Panic())
 	})
 
 	It("should correctly clone", func() {
 		ts.SetTransientState(bob, key, value)
-		Expect(ts.PeekAt(0).Get(bob, key), common.Hash{})
-		Expect(ts.GetTransientState(bob, key), value)
```
Comment (itsdevbear) [inline eth/core/state/journal/transient_storage_test.go:76]: @calbera I think you had a bug in your test that was being hidden by the incorrect `Expect` usage. Please verify the correctness here.

Comment (calbera) [reply eth/core/state/journal/transient_storage_test.go:76]: yeah this is the correct replacement, thanks @mmsqe for the fix!

**Review by itsdevbear — APPROVED** [review]

**Review by itsdevbear — DISMISSED** [review]

**Review by calbera — APPROVED** [review]

**Review by itsdevbear — APPROVED** [review]
lgtm, ci is just flake, we have to fix this geth p2p port issue

**Review by itsdevbear — APPROVED** [review]

**Comment by itsdevbear** [comment]
@archbear can you look at docker failures, stupid p2p port again, i might just rip it out today

**Review by ocnc — DISMISSED** [review]
lgtm

**Comment by itsdevbear** [comment]
@archbear 's pr should fix ci here

**Review by calbera — APPROVED** [review]

**Pull request merged**